File: DANTEAIServer/models/model_client.py
from app.extensions import db
from datetime import datetime

class Client(db.Model):
    __tablename__ = 'clients'

    id = db.Column(db.String(36), primary_key=True)
    company_id = db.Column(db.String(36), db.ForeignKey('companies.id_company'), nullable=False)

    name = db.Column(db.String(100), nullable=False)
    email = db.Column(db.String(120), unique=True, nullable=False)
    phone = db.Column(db.String(20))
    address = db.Column(db.String(255))
    document_type = db.Column(db.String(20))  # e.g., DNI, RIF, Passport
    document_number = db.Column(db.String(50), unique=True)

    avatar = db.Column(db.String(255))  # <- Nuevo campo para la imagen (ruta/nombre)

    is_active = db.Column(db.Boolean, default=True)

    created_at = db.Column(db.DateTime, default=datetime.utcnow)
    updated_at = db.Column(db.DateTime, onupdate=datetime.utcnow)

    # Relaciones
    company = db.relationship('Company', backref=db.backref('clients', lazy=True))

File: DANTEAIAPP/lib/core/models/model_client.dart
import 'dart:convert';

class ClientModel {
  final String id;
  final String companyId;
  final String name;
  final String email;
  final String? phone;
  final String? address;
  final String? documentType;
  final String? documentNumber;
  final bool isActive;
  final DateTime? createdAt;
  final DateTime? updatedAt;
  final String? avatar; // <-- nuevo campo

  ClientModel({
    required this.id,
    required this.companyId,
    required this.name,
    required this.email,
    this.phone,
    this.address,
    this.documentType,
    this.documentNumber,
    this.isActive = true,
    this.createdAt,
    this.updatedAt,
    this.avatar, // <-- nuevo campo en constructor
  });

  factory ClientModel.fromJson(Map<String, dynamic> json) {
    return ClientModel(
      id: json['id'],
      companyId: json['company_id'],
      name: json['name'],
      email: json['email'],
      phone: json['phone'],
      address: json['address'],
      documentType: json['document_type'],
      documentNumber: json['document_number'],
      isActive: json['is_active'] ?? true,
      createdAt: json['created_at'] != null
          ? DateTime.parse(json['created_at'])
          : null,
      updatedAt: json['updated_at'] != null
          ? DateTime.parse(json['updated_at'])
          : null,
      avatar: json['avatar'], // <-- agregar aquí
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'company_id': companyId,
      'name': name,
      'email': email,
      'phone': phone,
      'address': address,
      'document_type': documentType,
      'document_number': documentNumber,
      'is_active': isActive,
      'created_at': createdAt?.toIso8601String(),
      'updated_at': updatedAt?.toIso8601String(),
      'avatar': avatar, // <-- agregar aquí
    };
  }

  static List<ClientModel> listFromJson(String body) {
    final List parsed = json.decode(body);
    return parsed.map((json) => ClientModel.fromJson(json)).toList();
  }
}
